Tests/DependencyInjection/AbstractDoctrineExtensionTest.php

```diff
@@ -927,6 +937,17 @@ public function testAddFilter(): void
         $this->assertCount(2, $entityManager->getFilters()->getEnabledFilters());
     }
 
+    public function testDisablingLazyGhostOnOrm3Throws(): void
+    {
+        if (! class_exists(ProxyFactory::class) || method_exists(ProxyFactory::class, 'resetUninitializedProxy')) {
```

Comment:
Can you split to two branches and indicate which branch applies for which ORM version via comment? I guess ProxyFactory does not exist for ORM 3.0 and resetUninitializedProxy method does not exist for certain combination of ORM and doctrine/common? We need this stuff to make it easier to remove these BC layers later.

Reply:
`ProxyFactory` exists on all supported ORM versions. We have a CI job that uninstalls the ORM before running the tests, so the first check makes sure we actually have an ORM. The method has been removed in ORM 3, so if it exists, we have ORM 2.

Reply:
I've done as you said. It should be clearer now.